src/eckit/parser/YAMLParser.cc

```diff
@@ -425,12 +427,26 @@ Value YAMLParser::parseNumber() {
 }
 
 
+static std::string ignoreChar(const std::string& s, const char ignore) {
+    if (s.find(ignore) == std::string::npos) {
+        return s;
+    }
+    std::string str = s;
+    str.erase(remove(str.begin(), str.end(), ignore), str.end());
+    return str;
+}
+
```

Comment:
This still introduces an unnecessary copy. We can note that at the point that ignoreChar is called, the original string is never required again. So we can take ownership of the string here.

static std::string ignoreChar(std::string&& s, const char ignore) {
    std::string str(std::move(s));
    s.erase(std::remove(s.begin(), s.end(), ignore), s.end());
    return s;
}

or you could make this simpler by just being a wrapper around erase

static void eraseChar(std::string& s, const char ignore) {
    s.erase(std::remove(s.begin(), s.end(), ignore), s.end());
    return s;
}

To be honest, if I were writing it myself, I might just use erase/remove myself directly, in the calling code...

Reply:
@simondsmart Thanks for the detailed feedback again. I'll take the recommendation on ignoreChar and inline it.